### PR TITLE
Add pre-trained pipelines for XLS-R models

### PR DESCRIPTION
**Pull request by nateanl**
The PR adds three `Wav2Vec2Bundle ` pipeline objects for XLS-R models:
- WAV2VEC2_XLSR_300M
- WAV2VEC2_XLSR_1B
- WAV2VEC2_XLSR_2B

All three models use layer normalization in the feature extraction layers, hence `_normalize_waveform` is set to `True`.

### CODE REVIEW

**Comment by nateanl** [comment]
Due to the large size of the models, I didn't include the integration tests. Here is the log of testing `get_model` locally:

```python
>>> from torchaudio.pipelines import WAV2VEC2_XLSR_2B
>>> model = WAV2VEC2_XLSR_2B.get_model()
Downloading: "https://download.pytorch.org/torchaudio/models/wav2vec2_xlsr_2b.pth" to /Users/nateanl/.cache/torch/hub/checkpoints/wav2vec2_xlsr_2b.pth
100%|███████████████████████████████████████████████████████████████████| 8.04G/8.04G [04:43<00:00, 30.5MB/s]
>>> from torchaudio.pipelines import WAV2VEC2_XLSR_1B
>>> model = WAV2VEC2_XLSR_1B.get_model()
Downloading: "https://download.pytorch.org/torchaudio/models/wav2vec2_xlsr_1b.pth" to /Users/nateanl/.cache/torch/hub/checkpoints/wav2vec2_xlsr_1b.pth
100%|███████████████████████████████████████████████████████████████████| 3.59G/3.59G [01:49<00:00, 35.1MB/s]
>>> from torchaudio.pipelines import WAV2VEC2_XLSR_300M
>>> model = WAV2VEC2_XLSR_300M.get_model()
Downloading: "https://download.pytorch.org/torchaudio/models/wav2vec2_xlsr_300m.pth" to /Users/nateanl/.cache/torch/hub/checkpoints/wav2vec2_xlsr_300m.pth
100%|███████████████████████████████████████████████████████████████████| 1.18G/1.18G [00:35<00:00, 35.8MB/s]
```

File: torchaudio/pipelines/_wav2vec2/impl.py [hunk 1]
```diff
@@ -1415,3 +1418,152 @@ def _get_state_dict(self, dl_kwargs):
...
+        "encoder_dropout": 0.0,
+        "encoder_layer_norm_first": True,
+        "encoder_layer_drop": 0.0,
+        "aux_num_out": None,
+    },
+    _model_type="Wav2Vec2",
+    _sample_rate=16000,
+    _normalize_waveform=True,
+)
+WAV2VEC2_XLSR_300M.__doc__ = """XLS-R model with 300 millions of parameters,
```
Comment (hwangjeff) [inline torchaudio/pipelines/_wav2vec2/impl.py:1455]:
```suggestion
WAV2VEC2_XLSR_300M.__doc__ = """XLS-R model with 300 million parameters,
```

File: torchaudio/pipelines/_wav2vec2/impl.py [hunk 2]
```diff
@@ -1415,3 +1418,152 @@ def _get_state_dict(self, dl_kwargs):
...
+*BABEL* :cite:`Gales2014SpeechRA`, and
+*VoxPopuli* :cite:`voxpopuli`) in 128 languages,
+not fine-tuned.
+
+Originally published by the authors of *XLS-R* :cite:`babu2021xls` under MIT License and
+redistributed with the same license.
+[`License <https://github.com/facebookresearch/fairseq/blob/30c912b73c0f88d41171879b2f03226a171004ef/LICENSE>`__,
+`Source <https://github.com/facebookresearch/fairseq/tree/30c912b73c0f88d41171879b2f03226a171004ef/examples/wav2vec/xlsr#xls-r>`__]
+
+Please refer to :py:class:`torchaudio.pipelines.Wav2Vec2Bundle` for the usage.
```
Comment (hwangjeff) [inline torchaudio/pipelines/_wav2vec2/impl.py:1469]:
```suggestion
Please refer to :py:class:`torchaudio.pipelines.Wav2Vec2Bundle` for usage details.
```

File: torchaudio/pipelines/_wav2vec2/impl.py [hunk 3]
```diff
@@ -1415,3 +1418,152 @@ def _get_state_dict(self, dl_kwargs):
...
+        "encoder_dropout": 0.0,
+        "encoder_layer_norm_first": True,
+        "encoder_layer_drop": 0.0,
+        "aux_num_out": None,
+    },
+    _model_type="Wav2Vec2",
+    _sample_rate=16000,
+    _normalize_waveform=True,
+)
+WAV2VEC2_XLSR_1B.__doc__ = """XLS-R model with 1 billion of parameters,
```
Comment (hwangjeff) [inline torchaudio/pipelines/_wav2vec2/impl.py:1505]:
```suggestion
WAV2VEC2_XLSR_1B.__doc__ = """XLS-R model with 1 billion parameters,
```

File: torchaudio/pipelines/_wav2vec2/impl.py [hunk 4]
```diff
@@ -1415,3 +1418,152 @@ def _get_state_dict(self, dl_kwargs):
...
+        "encoder_dropout": 0.0,
+        "encoder_layer_norm_first": True,
+        "encoder_layer_drop": 0.0,
+        "aux_num_out": None,
+    },
+    _model_type="Wav2Vec2",
+    _sample_rate=16000,
+    _normalize_waveform=True,
+)
+WAV2VEC2_XLSR_2B.__doc__ = """XLS-R model with 2 billions of parameters,
```
Comment (hwangjeff) [inline torchaudio/pipelines/_wav2vec2/impl.py:1554]:
```suggestion
WAV2VEC2_XLSR_2B.__doc__ = """XLS-R model with 2 billion parameters,
```

**Review by hwangjeff — APPROVED** [review]

**Comment by nateanl** [comment]
The 2 Billion model test will fail due to large size, will remove it from integration tests.
